### DISCUSSION TITLE
Help understanding Zigbee Low-level commands

### DISCUSSION THREAD

**Discussion by phedders** [arendst/Tasmota#10406] [Q&A] [answered]
I have a few Drayton Wiser TRV's (made by Schneider) which are ZB. I have them paired but they dont support any commands that Tasmota already knows about. Please could someone give me pointers on how to write low level commands - perhaps I can then add more examples to the docs:

There are some recognised datas such as temps and set point:
┆ ☀️ 21.3°C 🎯 19.5°C ⚙️ 53%

I want to be able to se the "OccupiedHeatingSetpoint" and enable/disable boost if that is possible, but I'm a ZB novice!

 20210104-232235 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","Cluster":513,"Read":[18,0,8,28],"ReadNames":{"OccupiedHeatingSetpoint":true,"LocalTemperature":true,"PIHeatingDemand":true,"SystemMode":true},"Endpoint":2,"LinkQuality":136}}}

That would seem to indicate what can be read... which is good so I tried things like:

ZbSend { "device":"0x6B3D", "endpoint":2, "cluster":513,"read":18}

But I seem to just get errors:
20210104-232655 tele/zbb1/RESULT {"ZbRouteError":{"ShortAddr":"0x6B3D","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}

Other info that seems to come out every minute is:

20210104-232235 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","Cluster":0,"Read":[0],"ReadNames":{"ZCLVersion":true},"Endpoint":1,"LinkQuality":139}}}
20210104-232237 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","FE03/0020":"ADC,1407,1411,2,213,-32768,2350,0,498,100,214,212,213","Endpoint":1,"LinkQuality":141}}}
20210104-232238 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","FE03/0020":"ALG,14,1,235,213,28,0,498,1,498,99,100","LastMessageLQI":null,"LastMessageRSSI":-17,"0B05/E001":0,"Endpoint":1,"LinkQuality":141}}}
20210104-232322 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","FE03/0020":"UI,BoostDown","Endpoint":1,"LinkQuality":123}}}
20210104-232322 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","0201!80":"02029E073C00","Endpoint":1,"LinkQuality":126}}}
20210104-232322 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","LocalTemperature":21.3,"PIHeatingDemand":99,"OccupiedHeatingSetpoint":19.5,"Endpoint":1,"LinkQuality":128}}}
20210104-232322 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","Cluster":513,"Read":[18,0,8,28],"ReadNames":{"OccupiedHeatingSetpoint":true,"LocalTemperature":true,"PIHeatingDemand":true,"SystemMode":true},"Endpoint":2,"LinkQuality":123}}}
20210104-232323 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","BatteryVoltage":3.2,"BatteryPercentage":100,"Endpoint":1,"LinkQuality":123}}}
20210104-232324 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","FE03/0020":"ADC,1407,1412,2,213,-32768,1950,0,336,100,214,212,213","Endpoint":1,"LinkQuality":139}}}
20210104-232325 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","FE03/0020":"ALG,15,1,195,213,-18,-162,336,2,660,53,100","LastMessageLQI":null,"LastMessageRSSI":-17,"0B05/E001":0,"Endpoint":1,"LinkQuality":139}}}

Would very much appreciate any hints!
Thanks

**Comment by s-hadinger** [comment]
Can you do `ZbInfo 0x6B3D` and paste back result.

I'm afraid the device uses a proprietary protocol: `ADC,1407,1412,2,213,-32768,1950,0,336,100,214,212,213` and `ALG,15,1,195,213,-18,-162,336,2,660,53,100` are definitely not Zigbee standard.

Also, it's strange that your TRV is asking the coordinator for values 

`20210104-232235 tele/zbb1/SENSOR {"ZbReceived":{"WISER-01":{"Device":"0x6B3D","Name":"WISER-01","Cluster":513,"Read":[18,0,8,28],"ReadNames":{"OccupiedHeatingSetpoint":true,"LocalTemperature":true,"PIHeatingDemand":true,"SystemMode":true},"Endpoint":2,"LinkQuality":136}}}`

It should be the other way around.

**Comment by phedders** [comment]
On Tue, 2021-01-05 at 07:26 -0800, s-hadinger wrote:
> ZbInfo 0x6B3D

16:39:41.034 CMD: ZbInfo 0x6B3D
16:39:41.045 MQT: tele/zbb1/6B3D/SENSOR = {"ZbInfo":{"WISER-
01":{"Device":"0x6B3D","Name":"WISER-
01","IEEEAddr":"0x14B457FFFEB679C8","ModelId":"iTRV","Manufacturer":"Sc
hneider
Electric","Endpoints":[1,2],"Config":["T01"],"ThSetpoint":0,"TempTarget
":20,"Temperature":20.3,"Reachable":true,"BatteryPercentage":100,"LastS
een":36,"LastSeenEpoch":1609861145,"LinkQuality":149}}}
16:39:41.048 MQT: stat/zbb1/RESULT = {"ZbInfo":"Done"}

Easier to read like this though! :)



Thanks!

**Comment by s-hadinger** [comment, marked as answer]
Checking with Zigbee2MQTT, this should be straightforward.

Can you try:
`ZbSenc {"Device":"WISER-01","Write":{"OccupiedHeatingSetpoint":20}}` to set the TRV to a target of 20.0°C

**Comment by phedders** [comment]
On Tue, 2021-01-05 at 07:49 -0800, s-hadinger wrote:
> Checking with Zigbee2MQTT, this should be straightforward.
> Can you try:
> ZbSenc {"Device":"WISER-01","Write":{"OccupiedHeatingSetpoint":20}}
> to set the TRV to a target of 20.0°C

Yes I tried that:

16:50:20.358 MQT: tele/zbb1/STATE = {"Time":"2021-01-
05T16:50:20","Uptime":"0T14:40:21","UptimeSec":52821,"Vcc":3.509,"Heap"
:26,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":2,"Wifi":
{"AP":2,"SSId":"W0W","BSSId":"82:2A:A8:11:21:C9","Channel":11,"RSSI":66
,"Signal":-67,"LinkCount":1,"Downtime":"0T00:00:03"}}
16:50:33.493 CMD: ZbSenc {"Device":"WISER-
01","Write":{"OccupiedHeatingSetpoint":20}}
16:50:33.501 MQT: stat/zbb1/RESULT = {"Command":"Unknown"}

**Comment by phedders** [comment]
On Tue, 2021-01-05 at 07:49 -0800, s-hadinger wrote:
> Checking with Zigbee2MQTT, this should be straightforward.
> Can you try:
> ZbSenc {"Device":"WISER-01","Write":{"OccupiedHeatingSetpoint":20}}
> to set the TRV to a target of 20.0°C
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

OOps I just spotted a typo! Fixed it:

16:52:23.894 CMD: ZbSend {"Device":"WISER-
01","Write":{"OccupiedHeatingSetpoint":20}}
16:52:23.908 MQT: stat/zbb1/RESULT = {"ZbSend":"Done"}

**Comment by phedders** [comment]
On Tue, 2021-01-05 at 15:52 +0000, Paul Hedderly wrote:
> On Tue, 2021-01-05 at 07:49 -0800, s-hadinger wrote:
> > Checking with Zigbee2MQTT, this should be straightforward.
> > Can you try:
> > ZbSenc {"Device":"WISER-01","Write":{"OccupiedHeatingSetpoint":20}}
> > to set the TRV to a target of 20.0°C
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
> 
> OOps I just spotted a typo! Fixed it:
> 
> 16:52:23.894 CMD: ZbSend {"Device":"WISER-
> 01","Write":{"OccupiedHeatingSetpoint":20}}
> 16:52:23.908 MQT: stat/zbb1/RESULT = {"ZbSend":"Done"}

...And it seems to be working! - Many thanks, you're a star. Not sure
what I was doing wrong last night now, since I am sure I tried doing
exactly that.

I'll write a little tutorial section that can be added to the wiki of
the lines "if you have zbinfo output like.... then you should be able
to do...." highlighting what leads to the right resolution. The per
device docs on Z2M are very helpful too.

**Reply by s-hadinger** [reply]
ZbInfo was just to check the ModelId and Manufacturer and see in the Z2M code what clusters it is using.
